### ISSUE TITLE
Can't login verdaccio 3.3.0

### ISSUE THREAD

**Issue by BertrandGouny** [Alexandre-io/verdaccio-ldap#43]
Hello,
i'm unable to login with verdaccio 3.3.0 : 
```
info &lt;-- 10.244.73.1 requested 'POST /-/v1/login'
 http &lt;-- 404, user: undefined(XX.XX.XX.XX via 10.244.73.1), req: 'POST /-/v1/login', bytes: 21/150
 http &lt;-- 404, user: undefined(XX.XX.XX.XX via 10.244.73.1), req: 'POST /-/v1/login', bytes: 21/150
 info &lt;-- 10.244.73.1 requested 'PUT /-/user/org.couchdb.user:user.test'
 error--- unexpected error: plugin[method] is not a function
TypeError: plugin[method] is not a function
    at next (/usr/local/app/build/lib/auth.js:108:23)
    at Auth.add_user (/usr/local/app/build/lib/auth.js:118:7)
    at /usr/local/app/build/api/endpoint/api/user.js:24:12
    at Layer.handle [as handle_request] (/usr/local/app/node_modules/express/lib/router/layer.js:95:5)
    at next (/usr/local/app/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/usr/local/app/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/usr/local/app/node_modules/express/lib/router/layer.js:95:5)
    at /usr/local/app/node_modules/express/lib/router/index.js:281:22
    at param (/usr/local/app/node_modules/express/lib/router/index.js:354:14)
    at param (/usr/local/app/node_modules/express/lib/router/index.js:365:14)
    at param (/usr/local/app/node_modules/express/lib/router/index.js:365:14)
    at paramCallback (/usr/local/app/node_modules/express/lib/router/index.js:401:21)
    at /usr/local/app/build/api/middleware.js:36:7
    at paramCallback (/usr/local/app/node_modules/express/lib/router/index.js:404:7)
    at param (/usr/local/app/node_modules/express/lib/router/index.js:384:5)
    at Function.process_params (/usr/local/app/node_modules/express/lib/router/index.js:410:3)
 http &lt;-- 500, user: undefined(XX.XX.XX.XX via 10.244.73.1), req: 'PUT /-/user/org.couchdb.user:user.test', error: internal server error
(node:8) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
 http &lt;-- 500, user: undefined(XX.XX.XX.XX via 10.244.73.1), req: 'PUT /-/user/org.couchdb.user:user.test', error: internal server errorinfo &lt;-- 10.244.73.1 requested 'POST /-/v1/login'
 http &lt;-- 404, user: undefined(XX.XX.XX.XX via 10.244.73.1), req: 'POST /-/v1/login', bytes: 21/150
 http &lt;-- 404, user: undefined(XX.XX.XX.XX via 10.244.73.1), req: 'POST /-/v1/login', bytes: 21/150
 info &lt;-- 10.244.73.1 requested 'PUT /-/user/org.couchdb.user:user.test'
 error--- unexpected error: plugin[method] is not a function
TypeError: plugin[method] is not a function
    at next (/usr/local/app/build/lib/auth.js:108:23)
    at Auth.add_user (/usr/local/app/build/lib/auth.js:118:7)
    at /usr/local/app/build/api/endpoint/api/user.js:24:12
    at Layer.handle [as handle_request] (/usr/local/app/node_modules/express/lib/router/layer.js:95:5)
    at next (/usr/local/app/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/usr/local/app/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/usr/local/app/node_modules/express/lib/router/layer.js:95:5)
    at /usr/local/app/node_modules/express/lib/router/index.js:281:22
    at param (/usr/local/app/node_modules/express/lib/router/index.js:354:14)
    at param (/usr/local/app/node_modules/express/lib/router/index.js:365:14)
    at param (/usr/local/app/node_modules/express/lib/router/index.js:365:14)
    at paramCallback (/usr/local/app/node_modules/express/lib/router/index.js:401:21)
    at /usr/local/app/build/api/middleware.js:36:7
    at paramCallback (/usr/local/app/node_modules/express/lib/router/index.js:404:7)
    at param (/usr/local/app/node_modules/express/lib/router/index.js:384:5)
    at Function.process_params (/usr/local/app/node_modules/express/lib/router/index.js:410:3)
 http &lt;-- 500, user: undefined(XX.XX.XX.XX via 10.244.73.1), req: 'PUT /-/user/org.couchdb.user:user.test', error: internal server error
(node:8) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
 http &lt;-- 500, user: undefined(XX.XX.XX.XX via 10.244.73.1), req: 'PUT /-/user/org.couchdb.user:user.test', error: internal server error
```
config : 
```
auth:
  ldap:
    type: ldap
    groupNameAttribute: 'cn'
    client_options:
      url: "ldaps://openldap"
      adminDn: "cn=admin,dc=example,dc=org"
      adminPassword: "admin"
      searchBase: "ou=Users,dc=example,dc=org"
      searchFilter: "(&(uid={{username}})(memberOf=cn=Npm,ou=Groups,dc=example,dc=org))"
      groupDnProperty: 'cn'
      groupSearchBase: 'ou=Groups,dc=example,dc=org'
      searchAttributes: ['*', 'memberOf']
      cache: true
      reconnect: true
      tlsOptions:
        #ca: /verdaccio/tls/ldap/ca.crt
        #cert: /verdaccio/tls/ldap/cert.crt
        #key: /verdaccio/tls/ldap/cert.key
        rejectUnauthorized: False
```

guess plugin called function signature changed ? :s 

maybe this can help us : https://github.com/verdaccio/verdaccio/blob/master/test/flow/plugins/auth/example.auth.plugin.js

**Comment by BertrandGouny** [comment]
i try to login with npm command,
login in verdaccio UI works fine.

**Comment by moeterich** [comment]
Same error here. Web UI login works, npm login raises the error above.

**Comment by johannespfeiffer** [comment]
Might be related to https://github.com/verdaccio/verdaccio/pull/854

**Comment by Alexandre-io** [comment]
Verdaccio just released (3.4.0) a fix :)

**Comment by BertrandGouny** [comment]
thanks everyone,
i confirm this is fixed in 3.4.0
